src/parser/parser.yy

```diff
@@ -167,6 +168,31 @@ class GraphScanner;
 
 %%
 
+name_label
+     : LABEL { $$ = $1; }
+     | unreserved_keyword { $$ = $1; }
+     ;
+
+unreserved_keyword
+     : KW_SPACE              { $$ = new std::string("SPACE"); }
+     | KW_TAG                { $$ = new std::string("TAG"); }
+     | KW_VERTEX             { $$ = new std::string("VERTEX"); }
+     | KW_EDGE               { $$ = new std::string("EDGE"); }
```

Comment:
IMO. We should reserve `TAG`, `VERTEX` and `EDGE`, since they are widely used in our data plane queries, in which case we care about performance.

Reply:
Yeah, that makes sense.